Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -266,7 +271,211 @@ public override void DisconnectALMServer()
+                    }
+
+                    if (result == string.Empty)
+                    {
+                        result = "Export performed successfully.";
+                        return true;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export execution details to QC/ALM", ex);
+                return false;
+            }
+
+            return false;
+        }
+
+        private bool AddAttachment(string testSuiteId, string zipFileName)
+        {
+            try
+            {
+                FileStream fs = new FileStream(@"C:\Users\mkale\Desktop\ccd.pdf", FileMode.Open, FileAccess.Read);
+                BinaryReader br = new BinaryReader(fs);
+                byte[] fileData = br.ReadBytes((Int32)fs.Length);
+                var tt = Task.Run(() => { return entityService.AttachToEntity(new WorkspaceContext(1001, 4002), new TestSuite() { Id = new EntityId(testSuiteId) }, zipFileName.Split(Path.DirectorySeparatorChar).Last(), fileData, "text/zip", null); }).Result;
+                fs.Close();
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private RunSuite CreateRunSuite(PublishToALMConfig publishToALMConfig, BusinessFlow bizFlow, QCTestSet testSet, ObservableList<ExternalItemFieldBase> runFields)
+        {
+            try
+            {
+                RunSuite runSuiteToExport = new RunSuite();
+                runSuiteToExport.SetValue("subtype", "run_manual");
+                runSuiteToExport.Name = testSet.Name;
+                runSuiteToExport.SetValue("test", new BaseEntity()
+                {
+                    TypeName = "test_suite",
+                    Id = testSet.Id,
+                });
+                runSuiteToExport.SetValue("native_status", new BaseEntity()
+                {
+                    TypeName = "list_node",
+                    Id = "list_node.run_native_status." + bizFlow.RunStatus,
+                });
+                AddEntityFieldValues(runFields, runSuiteToExport, "test_suite");
+                //runSuiteToExport.SetValue("release", new BaseEntity()
+                //{
+                //    TypeName = "release",
+                //    Id = GetReleases().FirstOrDefault().Id
+                //});
+                return Task.Run(() => { return this.octaneRepository.CreateEntity<RunSuite>(GetLoginDTO(), runSuiteToExport, null); }).Result;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.DEBUG, "In CreateRunSuite/OctaneCore.cs method ", ex);
+                throw ex;
```

Comment:
Codacy found an issue: [Consider using 'throw;' to preserve the stack trace.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6361231)